plugins/out_lib/out_lib.c

```diff
@@ -211,6 +210,22 @@ static int out_lib_exit(void *data, struct flb_config *config)
     return 0;
 }
 
+static struct flb_config_map config_map[] = {
+    {
+     FLB_CONFIG_MAP_STR, "format", FLB_OUT_LIB_FMT_MSGPACK,
+     0, FLB_FALSE, 0,
```

Comment:
by setting here in the second and third arguments of the line to `FLB_FALSE, 0`, it means the config map will only validate it as a property but it won't write the value given in the context. e.g: `ctx->max_records` won't be updated.

Reply:
It is intentional.
Config parameter "format" is a string type for user convenience.
However the context has a field `format` and it is a int type to prevent comparing string at flush callback.

It is similar to "format" of out_file.
https://github.com/fluent/fluent-bit/blob/v2.2.2/plugins/out_file/file.c#L654

Reply:
my mistake, I wanted to refer to `max_records`, it's always zero